Classes/common/CDTIncrementalStore.m

```diff
@@ -0,0 +1,2798 @@
+{
+    self.isRelationship = NO;
+    NSAttributeType type = attribute.attributeType;
+    self.typeCode = type;
+    self.versionHash = attribute.versionHash;
+
+    switch (type) {
+        case NSUndefinedAttributeType:
+            self.typeName = @"NSUndefinedAttributeType";
+            break;
+
+        case NSStringAttributeType:
+            self.typeName = kCDTISStringAttributeType;
+            break;
+
+        case NSBooleanAttributeType:
+            self.typeName = kCDTISBooleanAttributeType;
+            break;
+
+        case NSDateAttributeType:
+            self.typeName = kCDTISDateAttributeType;
+            break;
+
+        case NSBinaryDataAttributeType:
+            self.typeName = kCDTISBinaryDataAttributeType;
+            break;
+
+        case NSTransformableAttributeType:
+            self.typeName = kCDTISTransformableAttributeType;
+            self.xform = [attribute valueTransformerName];
+            break;
+
+        case NSObjectIDAttributeType:
+            self.typeName = kCDTISObjectIDAttributeType;
+            break;
+
+        case NSDecimalAttributeType:
+            self.typeName = kCDTISDecimalAttributeType;
+            break;
+
+        case NSDoubleAttributeType:
+            self.typeName = kCDTISDoubleAttributeType;
+            break;
+
+        case NSFloatAttributeType:
+            self.typeName = kCDTISFloatAttributeType;
+            break;
+
+        case NSInteger16AttributeType:
+            self.typeName = kCDTISInteger16AttributeType;
+            break;
+
+        case NSInteger32AttributeType:
+            self.typeName = kCDTISInteger32AttributeType;
+            break;
+
+        case NSInteger64AttributeType:
+            self.typeName = kCDTISInteger64AttributeType;
+            break;
+
+        default:
+            return nil;
+    }
+    return self;
+}
+
+- (CDTISProperty *)initWithRelationship:(NSRelationshipDescription *)relationship
+{
+    self.isRelationship = YES;
+    self.versionHash = relationship.versionHash;
+
+    NSEntityDescription *ent = [relationship destinationEntity];
+    self.destination = [ent name];
+
+    if (relationship.isToMany) {
+        self.typeName = kCDTISRelationToManyType;
+        self.typeCode = CDTISRelationToManyType;
+    } else {
+        self.typeName = kCDTISRelationToOneType;
+        self.typeCode = CDTISRelationToOneType;
+    }
+    return self;
+}
+
+- (NSDictionary *)dictionary
+{
+    NSMutableDictionary *dic = [NSMutableDictionary dictionary];
+    dic[kCDTISPropertyNameKey] = self.typeName;
+    dic[kCDTISVersionHashKey] = stringFromData(self.versionHash);
+
+    if (self.xform) {
+        dic[kCDTISTransformerClassKey] = self.xform;
+    }
+    if (self.isRelationship && self.destination) {
+        dic[kCDTISRelationDesitinationKey] = self.destination;
+    }
+    return [NSDictionary dictionaryWithDictionary:dic];
+}
+
+- (NSInteger)typeCodeFromName:(NSString *)name
```

Comment:
ACK
My preference is to actually make convert methods that do not reference to object as a static C function.  This would be consistent with the rest of the code.  I personally find calling class methods cumbersome. @mkistler cool with that?